---
title: "Hypoxic Gene Expression"
author: "Jannik Hannemann"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  ioslides_presentation:
    incremental: true
    widescreen: true 
    logo: University-of-Seville.png
---

\newpage

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r include=FALSE}
library(tidyverse)
library(conflicted)
library(ggplot2)
library(MKinfer)
conflicts_prefer(dplyr::filter)
conflicts_prefer(dplyr::lag)
```
## Contenido
- Introducción sobre el proyecto
- Introducción sobre las herramientas utilizadas
- Análisis del codigo
- Análisis de los gráficos
- Resultados y conclusión

# Introducción
## Introducción sobre el proyecto
### **Objetivos:**
- Selección de genes de referencia **estables** es crucial para qRT-PCR
- Conjunto de datos de valores CT de 4 líneas celulares, 9 genes y 3 condiciones experimentales
- Encontrar los genes más estables y distinguir la influencia de las líneas celulares y las condiciones experimentales
- **Objetivo final:**
  - Representación gráfica de las varianzas en los valores CT para cada gen según las líneas celulares y las condiciones experimentales

## Introducción sobre las herramientas utilizadas
- R: Lenguaje de programación y entorno para análisis estadístico
- RStudio: IDE para R
- GitHub: Plataforma para control de versiones y colaboración en proyectos de programación
- tidyverse: Colección de paquetes para manipulación y visualización de datos
- ggplot2: Paquete para crear gráficos
- conflicted: Manejo de conflictos entre funciones de diferentes paquetes
- MKinfer: Paquete para inferencia estadística

# Analisis del código
## Primer contacto con los datos
```{r message=FALSE}
data <- read_csv2("GeneDataAllCellLinesNewFormat.csv")
#change the variable names to a usable format in R
names(data)<-str_replace_all(names(data), c(" " = "."))
knitr::kable(data[1:4,])
```

## Primer resumen de los datos
```{r message=FALSE}
data %>%
  group_by(Cell.Line, Gene, Experimental.Condition) %>% 
  summarise(mean(CT.Value, na.rm = TRUE)) %>% 
  head(n = 4) %>%
  knitr::kable()
```

## Obtener la varianza de los datos por línea celular
```{r echo=FALSE}
#get the data that is affected by the different cell lines
varianceDataCellLine <- data %>%
  #group the different "sets" of data
  group_by(Cell.Line, Gene, Experimental.Condition) %>% 
  #filter the data to only get the normoxic data, which means it will not be affected
  #by the experimental condition
  filter(Experimental.Condition == "24h NX") %>% 
  #group the data by the different genes
  group_by(Gene) %>% 
  summarise(
    #calculate the standard deviation of the CT values for each gene
    standardDeviationByCellLine = sd(CT.Value, na.rm = TRUE), 
    #calculate the mean of the CT values for each gene
    meanByCellLine = mean(CT.Value, na.rm = TRUE), 
    #calculate the coefficient of variation for each gene
    coefficientOfVariationCellLine = standardDeviationByCellLine / meanByCellLine, 
    #calculate the minimum of the confidence interval for the coefficient of variation
    confIntMinCellLine = sdCI(CT.Value)$conf.int[1] / meanByCellLine, 
    #calculate the maximum of the confidence interval for the coefficient of variation
    confIntMaxCellLine = sdCI(CT.Value)$conf.int[2] / meanByCellLine 
  )
knitr::kable(varianceDataCellLine[1:6, 1:5])
```

## Obtener la varianza de los datos por condiciones experimentales
```{r echo=FALSE}
#get the data that is affected by the different Experimental Conditions
varianceDataExpCond <- data %>%
  #group the different "sets" of data
  group_by(Cell.Line, Gene) %>% 
  #first get the data in groups for each different cell line,
  #because we want to clear out the effect of the different cell lines
  mutate(
    #calculate the standard deviation of the CT values for each gene and each cell line
    standardDeviationByExpCond = sd(CT.Value, na.rm = TRUE), 
    #calculate the mean of the CT values for each gene and each cell line
    meanByExpCond = mean(CT.Value, na.rm = TRUE), 
    #calculate the coefficient of variation for each gene and each cell line
    coefficientOfVariationExpCond = standardDeviationByExpCond / meanByExpCond, 
    #calculate the minimum of the confidence interval for the coefficient of variation
    confIntMinExpCond = sdCI(CT.Value)$conf.int[1] / meanByExpCond, 
    #calculate the maximum of the confidence interval for the coefficient of variation
    confIntMaxExpCond = sdCI(CT.Value)$conf.int[2] / meanByExpCond 
  ) %>% 
  #group the data by the different genes to get the means of all the values
  #for each gene (like this we clear out the effect of the different cell lines)
  group_by(Gene) %>% 
  summarise(
    #calculate the mean of the coefficient of variation for each gene
    coefficientOfVariationExpCond = mean(coefficientOfVariationExpCond, na.rm = TRUE), 
    #calculate the minimum of the confidence interval for the coefficient of variation
    confIntMinExpCond = mean(confIntMinExpCond, na.rm = TRUE), 
    #calculate the maximum of the confidence interval for the coefficient of variation
    confIntMaxExpCond = mean(confIntMaxExpCond, na.rm = TRUE) 
  )
knitr::kable(head(varianceDataExpCond))
```
## Juntar datos obtenidos
```{r message=FALSE}
finalData <- full_join(varianceDataCellLine, varianceDataExpCond, by = "Gene")
knitr::kable(finalData[1:5, 1:5])
```

## Graficar los datos
```{r}
ggplot(data = finalData) +
  aes(x = coefficientOfVariationExpCond, y = coefficientOfVariationCellLine, color = Gene) +
  geom_point()
```

## Añadir barras de error
```{r, echo=FALSE}
ggplot(data = finalData) +
  aes(x = coefficientOfVariationExpCond, y = coefficientOfVariationCellLine, color = Gene) +
  geom_point() +
  geom_errorbar(aes(xmin = confIntMinExpCond, xmax=confIntMaxExpCond), width=.01) +
  geom_errorbar(aes(ymin = confIntMinCellLine, ymax=confIntMaxCellLine), width=.01)
```

## Optimizar el gráfico (las barras de error son demasiado densas para una representación gráfica)
```{r, echo=FALSE}
ggplot(data = finalData) +
  aes(
    x = coefficientOfVariationExpCond * 100,
    y = coefficientOfVariationCellLine * 100,
    color = Gene
  ) +
  geom_point() +
  geom_text(label=finalData$Gene, hjust=0, vjust=0) +
  lims(x = c(0, 7.5), y = c(0, 7.5)) +
  #add a dashed line to show the 1:1 line.
  #Any gene that is under this line means that the
  #coefficient of variation of this gene is higher by
  #the experimental conditions than by the cell lines
  geom_abline(intercept = 0, slope = 1, color = "grey", linetype = "dashed") +
  labs(
    x = "The coefficient of variation by experimental conditions in %",
    y = "The coefficient of variation by cell lines in %",
    title = "Coefficient of variation of the CT-Values for the different analyzed genes"
  )
```